src/alternative-web/src/components/Pagination.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { usePropertyContext } from '../context/PropertyContext';
+
+const Pagination: React.FC = () => {
+  const { state, dispatch } = usePropertyContext();
+  const itemsPerPage = 2500;
```

Comment:
Ah yeh that be why… cheating a little there

Reply:
Ah that's fine haha. It's a big improvement of the existing pagination and at least all the images don't render at once.